### PR TITLE
Replace in-code model definition with yaml files

### LINKED ISSUE #19: Load neon model definitions from yaml files
Load the models from yaml instead of using in-code model definitions.

### PR DESCRIPTION
**Pull request by tmbo**
Fixes #19

- [x] ready for review

### CODE REVIEW

**Comment by duchoang** [comment]
:+1: 

The file `*.yaml` only contains layer info e.g. `1200 250 100 3` that the ios app needs. But we still need provide the `model.label.txt` to ios app.

So I am not sure is it worth using this complex file yaml in ios app to parse & extract only `1200 250 100 3` instead of the current simple one line `layer.txt`?

**Comment by janm399** [comment]
I'm actually OK with exporting the labels in labels.txt; the layers are OK at the moment, but doesn't the iOS app need the activation functions and biases? We can stick with the simple text files for the moment, but look to move to the YAML files in the future.

Also, having the YAML files will allow us to write a program that can help determine the best shape of the MLP given some restrictions. I'm thinking something along the lines of _we paid for 1 hour of AWS runtime, we might as well spend 55 minutes running the instances, and take the best result_, where best result is the result that has the lowest misclassification rate.

**Comment by tmbo** [comment]
You are right, the activation functions are hard coded right now (but to be fair they won't change anytime soon since they are the standard for MLPs).

We should definitely keep that in mind for the future. Hyperparameter optimization will get es a couple of percentage points on the misclassification scale. But since the trained model gets down o 0% on the test set for the exercise classification task, we need a harder challenge first (e.g. more exercises, people who do the exercises i really wird ways).

**Comment by janm399** [comment]
Actually, this is part of my _grand scheme_ for future work, where we have some infrastructure code for HP tuning. Let's have a separate issue with low priority.

**Comment by janm399** [comment]
I think this might have been a terrible idea on my part: the output layer depends on the number of examples, and I think it's a bit strange to have to have different YML files for different training sets.

**Comment by tmbo** [comment]
Yeah I noticed that as well. I do think it is a better approach, but that bugs me as well. We could change the layers node count in code to always make sure it fits (although this feels hacky since we are not honoring the yaml).
